src/page-objects/storefront/AccountOrder.ts

```diff
@@ -6,12 +6,55 @@ export class AccountOrder implements PageObject {
     public readonly cartLineItemImages: Locator;
     public readonly orderExpandButton: Locator;
     public readonly digitalProductDownloadButton: Locator;
+    public readonly dialogOrderCancel: Locator;
+    public readonly dialogOrderCancelButton: Locator;
+    public readonly dialogBackButton: Locator;
 
     constructor(public readonly page: Page) {
         this.orderExpandButton = page.getByRole('button', {name: /Expand|Show details/}).first();
         this.cartLineItemImages = page.locator('.line-item-img-link');
         this.digitalProductDownloadButton = page.getByRole('link', { name: 'Download' }).first();
+        this.dialogOrderCancel = page.getByRole('dialog', { name: 'Cancel order' });
+        this.dialogOrderCancelButton = this.dialogOrderCancel.getByRole('button', { name: 'Cancel order' });
+        this.dialogBackButton = this.dialogOrderCancel.getByRole('button', { name: 'Back' });
+
+    }
+    async getOrderByOrderNumber(orderNumber: string): Promise<Record<string, Locator>> {
+
+        const orderItem = this.page.getByRole('listitem').getByLabel(`Order number ${orderNumber}`);
+        const orderStatus = orderItem.locator('.order-table-header-order-status');
+        const orderHeading = orderItem.locator('.order-table-header-heading');
+        const orderActionsButton = orderItem.getByLabel('Actions');
+        const orderCancelButton = orderItem.getByRole('button', { name: 'Cancel order' });
+        const orderRepeatButton = orderItem.getByRole('button', { name: 'Repeat order' });
+        const orderChangePaymentMethodButton = orderItem.getByRole('link', { name: 'Change payment method' });
+        const orderShippingStatus = orderItem.getByLabel('Shipping status: ');
+        const orderPaymentStatus = orderItem.getByLabel('Payment status: ');
+        const orderPaymentMethod = orderItem.getByLabel('Payment method: ');
+        const orderShippingMethod = orderItem.getByLabel('Shipping method: ');
```

Comment:
Why do you have spaces at the end?

Reply:
it is defined in the platform code. Maybe it works also without it.

Reply:
I checked again and the A11Y is not so proper and you found a bug. Therefore I need to use the `nth()` method.